src/CommunityToolkit.Maui/Popup/StackBasedPopupLifecycleController.cs

```diff
@@ -0,0 +1,43 @@
+using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+
+namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// <see cref="Stack{T}"/> based implementation that manages the presentation of popups.
+/// </summary>
+public class StackBasedPopupLifecycleController : IPopupLifecycleController
+{
+	readonly Stack<WeakReference<Popup>> popupDisplayStack = new();
+
+	/// <inheritdoc cref="IPopupLifecycleController.GetCurrentPopup"/>
+	public Popup? GetCurrentPopup()
+	{
+		var popupReference = popupDisplayStack.Peek();
+
+		if (popupReference.TryGetTarget(out var popup))
+		{
+			return popup;
+		}
+
+		return null;
+	}
+
+	/// <inheritdoc cref="IPopupLifecycleController.OnShowPopup"/>
+	public void OnShowPopup(Popup popup)
+	{
+		popup.Closed += OnPopupClosed;
+		popupDisplayStack.Push(new(popup));
+	}
+
+	void OnPopupClosed(object? sender, PopupClosedEventArgs e)
+	{
+		if (sender is not Popup popup)
+		{
+			return;
+		}
+
+		popup.Closed -= OnPopupClosed;
+		popupDisplayStack.Pop();
```

Comment:
Are we sure we can close only the top popup?
With the popup Close method, user can close any popup in stack.

Reply:
Initially I thought we could ship with this Stack based implementation and then should the need arise to support more complex scenarios we could introduce a new PopupLifecycleController.

I think iOS limits us to being stack based though

Reply:
I have opted for a list based approach now and look up the popup being closed 👍